hazelcast/src/main/java/com/hazelcast/internal/partition/operation/MigrationRequestOperation.java

```diff
@@ -200,14 +171,8 @@ private void trySendNewFragment() {
 
             // migration invocation must always happen on partition thread
             if (migrationState != null) {
-                // RU_COMPAT_4_2
-                if (getNodeEngine().getClusterService().getClusterVersion().isLessThan(Versions.V5_0)) {
-                    invokeMigrationOperation(migrationState, false);
-                } else {
-                    // this is executed in async pool thread
```

Comment:
Does removing this line of comment intentional?

Reply:
This comment seemed superfluous because the 4.2 compatibility code is gone and there is no way that `trySendNewFragment` is executed from partition thread -- it is now always called from async thread. We already indicate that `trySendNewFragment` is executed on async executor thread at the [call site](https://github.com/hazelcast/hazelcast/pull/19618/files#diff-8ae4f7b6f47cac763ef48111f12148378915f37b7c8b0e5aa266b048336330f8R346-R349).